### ISSUE TITLE
cc65: Inline assembler doesn't accept stack-relative lda %o, S

### ISSUE THREAD

**Issue by semplar2007** [cc65/cc65#994]
Example:
```
void dma_setup(mode) {
  asm("lda %o, S", mode);
}
```
This is correct ca65 assembler code, but compiler doesn't allow to generate it.
Output is:
```
snes_tools.h(122): Error: ASM code error: syntax error
```

**Comment by semplar2007** [comment]
Oh I think cc65 supports only 6502 and 65c02 instructions, and Stack-relative adressing mode is 65816 instruction, so that's why it doesn't support that.

**Comment by greg-king5** [comment]
> Oh, I think cc65 supports only 6502 and 65c02 instructions; and, Stack-relative addressing mode is 65816 instruction; so that's why it doesn't support that.

Yes.  If we want to allow those extra addressing modes, then we must extend the optimizer, so that it can recognize and classify them.
